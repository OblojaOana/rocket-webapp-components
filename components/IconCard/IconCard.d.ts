export interface IconCardProps {

    /**
     * Icon to be displayed. This property is mandatory
     */
    icon: Object;

    /**
     * The color of the icon that will be displayed
     * @default "theme"
     */
    iconColor?: "orange" |
    "green" |
    "red" |
    "blue" |
    "purple" |
    "rose" |
    "theme"

    /**
     * The title to be displayed. This property can be a string or a component
     * @default ""
     */
    title?: React.ReactNode;

    /**
     * 
     */
    category?: React.ReactNode;

    /**
     * The content of the card
     */
    content?: React.ReactNode;

    /**
     * 
     */
    footer?: React.ReactNode;

    /**
     * If set to true, the background will be transparent
     */
    plain?: Boolean;

    /**
     * Override or extend the styles applied to the content of the card
     */
    customCardContentClass?: String;

    /**
     * Override or extend the styles applied to the card
     */
    customCardClass?: String;

    /**
     * Demos:
     * 
     * - https://bit.dev/totalsoft_oss/react-mui/icon-card
     * 
     */

}

export default function IconCard(props: IconCardProps): JSX.Element;