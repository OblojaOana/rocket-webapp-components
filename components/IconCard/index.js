import IconCard from "../../core/IconCard";
export default IconCard;