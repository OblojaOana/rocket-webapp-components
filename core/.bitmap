/* THIS IS A BIT-AUTO-GENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY. */

{
    "totalsoft_oss.react-mui/new/core@0.0.11": {
        "files": [
            {
                "relativePath": "src/core/index.js",
                "test": false,
                "name": "index.js"
            }
        ],
        "mainFile": "src/core/index.js",
        "trackDir": "src/core",
        "origin": "AUTHORED",
        "exported": true
    },
    "version": "14.8.8"
}