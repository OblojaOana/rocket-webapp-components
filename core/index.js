//Buttons
export { default as Button } from '@bit/totalsoft_oss.react-mui.button'
export { default as Card } from '@bit/totalsoft_oss.react-mui.card'
export { default as AddButton } from '@bit/totalsoft_oss.react-mui.add-button'
export { default as DeleteButton } from '@bit/totalsoft_oss.react-mui.delete-button'
export { default as UploadButton } from '@bit/totalsoft_oss.react-mui.upload-button'
export { default as BackToButton } from '@bit/totalsoft_oss.react-mui.back-to-button'
export { default as IconButton } from '@bit/totalsoft_oss.react-mui.icon-button'
export { default as CancelButton } from '@bit/totalsoft_oss.react-mui.cancel-button'
export { default as DownloadButton } from '@bit/totalsoft_oss.react-mui.download-button'
export { default as DownwardButton } from '@bit/totalsoft_oss.react-mui.downward-button'
export { default as EditButton } from '@bit/totalsoft_oss.react-mui.edit-button'
export { default as SaveButton } from '@bit/totalsoft_oss.react-mui.save-button'
export { default as UpwardButton } from '@bit/totalsoft_oss.react-mui.upward-button'

//Cards
export { default as IconCard } from '@bit/totalsoft_oss.react-mui.icon-card'
export { default as IconCollapseCard } from '@bit/totalsoft_oss.react-mui.icon-collapse-card'
export { default as RegularCard } from '@bit/totalsoft_oss.react-mui.regular-card'
export { default as ChartCard } from '@bit/totalsoft_oss.react-mui.chart-card'
export { default as StatsCard } from '@bit/totalsoft_oss.react-mui.stats-card'
export { default as CardTitle } from '@bit/totalsoft_oss.react-mui.card-title'
export { default as ImageCard } from '@bit/totalsoft_oss.react-mui.image-card'

//FakeText
export { default as LoadingFakeText } from '@bit/totalsoft_oss.react-mui.fake-text'

//Inputs
export { default as CustomTextField } from '@bit/totalsoft_oss.react-mui.custom-text-field'
export { default as DateTime } from '@bit/totalsoft_oss.react-mui.date-time'
export { default as DynamicInput } from '@bit/totalsoft_oss.react-mui.dynamic-input'

//NavPills
export { default as NavPills } from '@bit/totalsoft_oss.react-mui.nav-pills'

//Pagination
export { default as Pagination } from '@bit/totalsoft_oss.react-mui.pagination'

//AutoComplete  
export { default as Autocomplete } from '@bit/totalsoft_oss.react-mui.autocomplete'

//Forbidden and NotFound
export { default as Forbidden } from '@bit/totalsoft_oss.react-mui.forbidden'
export { default as NotFound } from '@bit/totalsoft_oss.react-mui.not-found'

//SimpleSlider
export { default as SimpleSlider } from '@bit/totalsoft_oss.react-mui.simple-slider'

//TimePicker
export { default as TimePicker } from '@bit/totalsoft_oss.react-mui.time-picker'

//ExpandingText
export { default as ExpandingText } from '@bit/totalsoft_oss.react-mui.expanding-text'

//Accordion
export { default as Accordion } from '@bit/totalsoft_oss.react-mui.accordion'
export { default as Piano } from '@bit/totalsoft_oss.react-mui.piano'

//Chart factory
export { default as ChartFactory } from '@bit/totalsoft_oss.react-mui.charts'

//CustomDialog
export { default as CustomDialog } from '@bit/totalsoft_oss.react-mui.custom-dialog'

//CustomLinearProgress
export { default as CustomLinearProgress } from '@bit/totalsoft_oss.react-mui.custom-linear-progress'

//Typography
export { default as Typography } from '@bit/totalsoft_oss.react-mui.typography'

//CheckInternetConnection
export { default as CheckInternetConnection } from '@bit/totalsoft_oss.react-mui.check-internet-connection'

//DialogDisplay
export { default as DialogDisplay } from '@bit/totalsoft_oss.react-mui.dialog-display'

//PasswordField
export { default as PasswordField } from '@bit/totalsoft_oss.react-mui.password-field'

//Toast
export { useToast, ToastContainer } from '@bit/totalsoft_oss.react-mui.toast'