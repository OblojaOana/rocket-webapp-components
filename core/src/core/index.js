// //Buttons
// export { default as Button } from './Button'
// export { default as AddButton } from './AddButton'
// export { default as DeleteButton } from './DeleteButton'
// export { default as UploadButton } from './UploadButton'
// export { default as BackToButton } from './BackToButton'
// export { default as IconButton } from './IconButton'
// export { default as CancelButton } from './CancelButton'
// export { default as DownloadButton } from './DownloadButton'
// export { default as DownwardButton } from './DownwardButton'
// export { default as EditButton } from './EditButton'
// export { default as SaveButton } from './SaveButton'
// export { default as UpwardButton } from './UpwardButton'

// //Cards
// export { default as IconCard } from './IconCard'
// export { default as IconCollapseCard } from './IconCollapseCard'
// export { default as RegularCard } from './RegularCard'
// export { default as ChartCard } from './ChartCard'
// export { default as StatsCard } from './StatsCard'
// export { default as CardTitle } from './CardTitle'
// export { default as ImageCard } from './ImageCard'

// //FakeText
// export { default as LoadingFakeText } from './FakeText'

// //Inputs
// export { default as CustomTextField } from './CustomTextField'
// export { default as DateTime } from './DateTime'
// export { default as DynamicInput } from './DynamicInput'

// //NavPills
// export { default as NavPills } from './NavPills'

// //Pagination
// export { default as Pagination } from './Pagination'

// //AutoComplete  
// export { default as Autocomplete } from './Autocomplete'

// //Forbidden and NotFound
// export { default as Forbidden } from './Forbidden/Forbidden'
// export { default as NotFound } from './NotFound/NotFound'

// //SimpleSlider
// export { default as SimpleSlider } from './SimpleSlider'

// //TimePicker
// export { default as TimePicker } from './TimePicker'

// //ExpandingText
// export { default as ExpandingText } from './ExpandingText'

// //Accordion
// export { default as Accordion } from './Accordion'
// export { default as Piano } from './Piano'

// //Chart factory
// export { default as ChartFactory } from './Charts'

// //CustomDialog
// export { default as CustomDialog } from './CustomDialog'

// //CustomLinearProgress
// export { default as CustomLinearProgress } from './CustomLinearProgress'

// //Typography
// export { default as Typography } from './Typography'

// //CheckInternetConnection
// export { default as CheckInternetConnection } from './CheckInternetConnection'

// //DialogDisplay
// export { default as DialogDisplay } from './DialogDisplay'

// //PasswordField
// export { default as PasswordField } from './PasswordField'

// //Toast
// export { useToast, ToastContainer } from './Toast'


import LoadingFakeText from '@bit/totalsoft_oss.react-mui.fake-text';
import SaveButton from '@bit/totalsoft_oss.react-mui.save-button';
import RegularCard from '@bit/totalsoft_oss.react-mui.regular-card';
import Piano from '@bit/totalsoft_oss.react-mui.piano';
import Toast from '@bit/totalsoft_oss.react-mui.toast';
import UpwardButton from '@bit/totalsoft_oss.react-mui.upward-button';
import DownwardButton from '@bit/totalsoft_oss.react-mui.downward-button';
import Forbidden from '@bit/totalsoft_oss.react-mui.forbidden';
import Card from '@bit/totalsoft_oss.react-mui.card';
import StatsCard from '@bit/totalsoft_oss.react-mui.stats-card';
import CustomDialog from '@bit/totalsoft_oss.react-mui.custom-dialog';
import UploadButton from '@bit/totalsoft_oss.react-mui.upload-button';
import Pagination from '@bit/totalsoft_oss.react-mui.pagination';
import TimePicker from '@bit/totalsoft_oss.react-mui.time-picker';
import PasswordField from '@bit/totalsoft_oss.react-mui.password-field';
import FileUploadButton from '@bit/totalsoft_oss.react-mui.file-upload-button';
import DynamicInput from '@bit/totalsoft_oss.react-mui.dynamic-input';
import Charts from '@bit/totalsoft_oss.react-mui.charts';
import NavPills from '@bit/totalsoft_oss.react-mui.nav-pills';
import DeleteButton from '@bit/totalsoft_oss.react-mui.delete-button';
import CustomLinearProgress from '@bit/totalsoft_oss.react-mui.custom-linear-progress';
import IconButton from '@bit/totalsoft_oss.react-mui.icon-button';
import CheckInternetConnection from '@bit/totalsoft_oss.react-mui.check-internet-connection';
import SimpleSlider from '@bit/totalsoft_oss.react-mui.simple-slider';
import ImageCard from '@bit/totalsoft_oss.react-mui.image-card';
import Button from '@bit/totalsoft_oss.react-mui.button';
import CustomTextField from '@bit/totalsoft_oss.react-mui.custom-text-field';
import BackToButton from '@bit/totalsoft_oss.react-mui.back-to-button';
import DateTime from '@bit/totalsoft_oss.react-mui.date-time';
import IconCollapseCard from '@bit/totalsoft_oss.react-mui.icon-collapse-card';
import DialogDisplay from '@bit/totalsoft_oss.react-mui.dialog-display';
import NotFound from '@bit/totalsoft_oss.react-mui.not-found';
import Typography from '@bit/totalsoft_oss.react-mui.typography';
import IconCard from '@bit/totalsoft_oss.react-mui.icon-card';
import EditButton from '@bit/totalsoft_oss.react-mui.edit-button';
import CardTitle from '@bit/totalsoft_oss.react-mui.card-title';
import CancelButton from '@bit/totalsoft_oss.react-mui.cancel-button';
import DownloadButton from '@bit/totalsoft_oss.react-mui.download-button';

export default { LoadingFakeText, SaveButton, RegularCard }