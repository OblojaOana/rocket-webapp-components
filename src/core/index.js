//Buttons
export { default as Button } from './Button'
export { default as AddButton } from './AddButton'
export { default as DeleteButton } from './DeleteButton'
export { default as UploadButton } from './UploadButton'
export { default as BackToButton } from './BackToButton'
export { default as IconButton } from './IconButton'
export { default as CancelButton } from './CancelButton'
export { default as DownloadButton } from './DownloadButton'
export { default as DownwardButton } from './DownwardButton'
export { default as EditButton } from './EditButton'
export { default as SaveButton } from './SaveButton'
export { default as UpwardButton } from './UpwardButton'

//Cards
export { default as IconCard } from './IconCard'
export { default as IconCollapseCard } from './IconCollapseCard'
export { default as RegularCard } from './RegularCard'
export { default as ChartCard } from './ChartCard'
export { default as StatsCard } from './StatsCard'
export { default as CardTitle } from './CardTitle'
export { default as ImageCard } from './ImageCard'

//FakeText
export { default as LoadingFakeText } from './FakeText'

//Inputs
export { default as CustomTextField } from './CustomTextField'
export { default as DateTime } from './DateTime'
export { default as DynamicInput } from './DynamicInput'

//NavPills
export { default as NavPills } from './NavPills'

//Pagination
export { default as Pagination } from './Pagination'

//AutoComplete  
export { default as Autocomplete } from './Autocomplete'

//Forbidden and NotFound
export { default as Forbidden } from './Forbidden/Forbidden'
export { default as NotFound } from './NotFound/NotFound'

//SimpleSlider
export { default as SimpleSlider } from './SimpleSlider'

//TimePicker
export { default as TimePicker } from './TimePicker'

//ExpandingText
export { default as ExpandingText } from './ExpandingText'

//Accordion
export { default as Accordion } from './Accordion'
export { default as Piano } from './Piano'

//Chart factory
export { default as ChartFactory } from './Charts'

export { default as CustomDialog } from './CustomDialog'

export { default as CustomLinearProgress } from './CustomLinearProgress'

export { default as Typography } from './Typography'

export { default as CheckInternetConnection } from './CheckInternetConnection'

export { default as DialogDisplay } from './DialogDisplay'

export { default as PasswordField } from './PasswordField'

//Toast
export { useToast, ToastContainer } from './Toast'