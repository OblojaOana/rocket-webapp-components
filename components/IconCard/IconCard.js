import React from "react";
import cx from "classnames";
import PropTypes from "prop-types";
import { Card, CardContent, CardHeader, makeStyles } from "@material-ui/core";
import iconCardStyle from "./iconCardStyle";
import Typography from "../Typography";

const useStyles = makeStyles(iconCardStyle);

const IconCard = props => {
  const classes = useStyles();
  const {
    title,
    content,
    iconColor,
    subtitle,
    footer,
    plain,
    customCardContentClass,
    customCardClass,
    icon: Icon
  } = props;
  const cardClasses =
    classes.card +
    " " +
    cx({
      [classes.cardPlain]: plain,
      [customCardClass]: customCardClass
    });
  const cardContentClasses =
    classes.cardContent +
    " " +
    cx({
      [customCardContentClass]: customCardContentClass !== undefined
    });

  return (
    <Card className={cardClasses}>
      <CardHeader
        classes={{
          root: classes.cardHeader + " " + classes[iconColor + "CardHeader"],
          avatar: classes.cardAvatar
        }}
        avatar={<Icon className={classes.cardIcon} />}
      />
      <CardContent className={cardContentClasses}>
        <Typography className={classes.cardTitle} variant="subtitle1">
          {title}
        </Typography>
        {subtitle && (
          <Typography className={classes.cardCategory} variant="subtitle2">
            {subtitle}
          </Typography>
        )}
        {content}
      </CardContent>
      {footer && <div className={classes.cardFooter}>{footer}</div>}
    </Card>
  );
};

IconCard.defaultProps = {
  iconColor: "theme",
  title: ""
};

IconCard.propTypes = {
  icon: PropTypes.object.isRequired,
  iconColor: PropTypes.oneOf([
    "orange",
    "green",
    "red",
    "blue",
    "purple",
    "rose",
    "theme"
  ]),
  title: PropTypes.node,
  subtitle: PropTypes.node,
  content: PropTypes.node,
  footer: PropTypes.node,
  plain: PropTypes.bool,
  customCardContentClass: PropTypes.string,
  customCardClass: PropTypes.string
};

export default IconCard;
