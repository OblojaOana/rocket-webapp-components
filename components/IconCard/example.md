##### Copy this snapshot in the overview panel

```js
import React from "react";
import IconCard from "@bit/totalsoft_oss.react-mui.icon-card";
import FastfoodIcon from '@material-ui/icons/Fastfood';

export default (
	<IconCard
		icon={FastfoodIcon}
		iconColor="theme"
		title="Shrimp and Chorizo Paella"
		content={<span>This impressive paella is a perfect party dish and a fun meal to cook together with your guests. Add 1 cup of frozen peas along with the mussels, if you like.</span>}
	/>
);
```


##### Copy this snapshot in the overview panel

```js
import React from "react";
import IconCard from "@bit/totalsoft_oss.react-mui.icon-card";
import Assignment from '@material-ui/icons/Assignment';
import CardTitle from "@bit/totalsoft_oss.react-mui.card-title";
import AddButton from "@bit/totalsoft_oss.react-mui.add-button";
import CustomTextField from "@bit/totalsoft_oss.react-mui.custom-text-field";
import { Table, Thead, Tbody, Tr, Th, Td } from 'react-super-responsive-table';
import { Grid } from "@material-ui/core";

export default (
	<IconCard
		icon={Assignment}
		iconColor="theme"
		title={
			<CardTitle
				title={"Documents"}
				actions={[<AddButton key='addButton' title={"Add document"} />]}
			/>
		}
		content={
			<Grid>
				<Table>
					<Thead>
						<Tr>
							<Th >{"File names"}</Th>
							<Th >{"Description"}</Th>
							<Th />
						</Tr>
					</Thead>
					<Tbody>
						<Tr>
							<Td >
								<CustomTextField
									fullWidth
									value={"File name"}
									onChange={() => { }}
								/>
							</Td>
							<Td>
								<CustomTextField
									fullWidth
									value={"Description"}
									onChange={() => { }}
								/>
							</Td>
						</Tr>
					</Tbody>
				</Table>
			</Grid>
		}
	/>
);
```