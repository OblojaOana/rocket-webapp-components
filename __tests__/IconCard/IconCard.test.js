import React from "react";
import renderer from "react-test-renderer";
import IconCard from "../../components/IconCard/IconCard";
import MailOutline from "@material-ui/icons/MailOutline";
import { mount } from 'enzyme';
import { Card, CardHeader } from "@material-ui/core";

describe("IconCard", () => {
  it("Snapshot", () => {
    const component = renderer.create(
      <IconCard
        icon={MailOutline}
        iconColor="theme"
        title="Title card"
        subtitle="Subtitle card"
        footer="Footer card"
        content={"Content of the card"}
      />
    );

    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it("classes cardStyle and textStyle should be present", () => {
    const wrapper = mount(
      <IconCard
        icon={MailOutline}
        iconColor="theme"
        title="Title card"
        subtitle="Subtitle card"
        footer="Footer card"
        content={"Content of the card"}
      />
    );

    expect(wrapper.find(CardHeader).props().classes.root).toContain('cardHeader')
    expect(wrapper.find(CardHeader).props().classes.avatar).toContain('cardAvatar')
  });

  it("if footer is provided then classFooter should be present", () => {
    const wrapper = mount(
      <IconCard
        icon={MailOutline}
        iconColor="theme"
        title="Title card"
        subtitle="Subtitle card"
        footer="Footer card"
        content={"Content of the card"}
      />
    );

    expect(wrapper.find(Card).find('div').last().props().className).toContain('cardFooter')
  });

  it("if footer is not provided then classFooter should not be present", () => {
    const wrapper = mount(
      <IconCard
        icon={MailOutline}
        iconColor="theme"
        title="Title card"
        subtitle="Subtitle card"
        content={"Content of the card"}
      />
    );

    expect(wrapper.find(Card).find('div').last().props().className).not.toContain('cardFooter')
  });
});
