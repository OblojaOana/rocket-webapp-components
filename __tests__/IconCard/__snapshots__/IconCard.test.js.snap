// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCard Snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1  MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root makeStyles-cardHeader-3 makeStyles-themeCardHeader-10"
  >
    <div
      className="MuiCardHeader-avatar makeStyles-cardAvatar-12"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-cardIcon-13"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
        />
      </svg>
    </div>
    <div
      className="MuiCardHeader-content"
    />
  </div>
  <div
    className="MuiCardContent-root makeStyles-cardContent-11 "
  >
    <h6
      className="MuiTypography-root makeStyles-cardTitle-14 MuiTypography-subtitle1"
    >
      Title card
    </h6>
    <h6
      className="MuiTypography-root makeStyles-cardCategory-15 MuiTypography-subtitle2"
    >
      Subtitle card
    </h6>
    Content of the card
  </div>
  <div
    className="makeStyles-cardFooter-16"
  >
    Footer card
  </div>
</div>
`;
